cmd/evicter/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+
+	v1 "k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/typed/policy/v1beta1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+func main() {
+	var (
+		kubeconfig                     = flag.String("kubeconfig", "", "absolute path to the kubeconfig file: `<home>/.kube/config`")
+		master                         = flag.String("master", "", "master url")
+		labelSelector                  = flag.String("label-selector", "tainted=true", "label selector to discover tainted pods")
+		terminationGracePeriodSeconds  = flag.Int64("termination-grace-period", 30, "pod termination grace period in seconds")
+		taintedIncubationPeriodSeconds = flag.Int64("incubation-period", 24*60*60, "time in seconds a tainted pod can run before eviction")
+	)
+	flag.Parse()
+	flag.Set("logtostderr", "true") // glog: no disk log
+
+	config, err := clientcmd.BuildConfigFromFlags(*master, *kubeconfig)
```

Comment:
If kubeconfig is not provided we should fall back to in-cluster config: https://github.com/kubernetes/client-go/blob/master/examples/in-cluster-client-configuration/main.go#L27-L41

If run as an in-cluster operator with a service account, it will use this method of connecting to the kube-apiserver.

Reply:
It falls back to `InClusterConfig` when args are empty. See https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L544 

The configuration parameters are supposed for local testing/ debugging when you want to run it against a real (test) cluster.